net/yggdrasil/files/yggdrasil.sh

```diff
@@ -114,45 +98,35 @@ proto_yggdrasil_setup() {
 	local node_info_privacy
 
 	config_load network
-	config_get private_key "${config}" "private_key"
-	config_get public_key "${config}" "public_key"
-	config_get mtu "${config}" "mtu"
-	config_get node_info "${config}" "node_info"
-	config_get node_info_privacy "${config}" "node_info_privacy"
+	config_get private_key       "$config" "private_key"
+	config_get public_key        "$config" "public_key"
+	config_get mtu               "$config" "mtu"
+	config_get node_info         "$config" "node_info"
+	config_get node_info_privacy "$config" "node_info_privacy"
 
-	if [ -z $private_key ]; then
-		proto_yggdrasil_generate_keypair
-	fi;
+	[ -z "$private_key" ] && proto_yggdrasil_generate_keypair
 
 	umask 077
-	mkdir -p "${ygg_dir}"
+	mkdir -p "$ygg_dir"
 
-	if [ $private_key = "auto" ]; then
+	if [ "$private_key" = "auto" ]; then
 		proto_yggdrasil_generate_keypair
-		uci -t ${ygg_dir}/.uci.${config} batch <<EOF
+		uci -t "${ygg_dir}/.uci.${config}" batch <<EOF
 			set network.${config}.private_key='${private_key}'
 			set network.${config}.public_key='${public_key}'
 EOF
-		uci -t ${ygg_dir}/.uci.${config} commit;
+		uci -t "${ygg_dir}/.uci.${config}" commit;
 	fi;
 
 	# Generate config file
 	json_init
-	json_add_string "IfName" ${config}
-	json_add_string "AdminListen" ${ygg_sock}
-
-	json_add_string "PrivateKey" ${private_key}
-	json_add_string "PublicKey" ${public_key}
-
-	if [ ! -z $mtu ]; then
-		json_add_int "IfMTU" ${mtu}
-	fi;
-
-	if [ ! -z $node_info ]; then
-		json_add_string "NodeInfo" "%%_YGGDRASIL_NODEINFO_TEMPLATE_%%"
-	fi;
-
-	json_add_boolean "NodeInfoPrivacy" ${node_info_privacy}
+	json_add_string "IfName" "$config"
+	json_add_string "AdminListen" "$ygg_sock"
+	json_add_string "PrivateKey" "$private_key"
+	json_add_string "PublicKey" "$public_key"
+	[ -n "$mtu" ] && json_add_int "IfMTU" "$mtu"
+	[ -n "$node_info" ] && json_add_string "NodeInfo" "%%_YGGDRASIL_NODEINFO_TEMPLATE_%%"
```

Comment:
yes, replacing all ! true is good but, i cant take any of the short circuit logic.

Reply:
You should at least use `-n` instead of `! -z`.